### PR TITLE
{Compute} `az vm`: Migrate commands to Code Gen V2

### PR DESCRIPTION
**Pull request by yanzhudd**
**Related command**

- `az vm deallocate/list-sizes/generalize/list-vm-resize-options`
- `az vm convert/start/reapply/redeploy/assess-patches/simulate-eviction/perform-maintenance`

**Description**

Migrate commands in `az vm` group:

- `az vm deallocate/list-sizes/generalize/list-vm-resize-options`
- `az vm convert/start/reapply/redeploy/assess-patches/simulate-eviction/perform-maintenance`

Some commands have not been migrated yet:

- `az vm capture/create/delete/get-instance-view/list/list-ip-addresses`
- `az vm list-skus/list-usage/open-port/resize/restart/show/stop/update/wait/auto-shutdown/install-patches`

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

**Comment by yanzhudd** [comment]
> ❌AzureCLI-BreakingChangeTest

It has been verified with no breaking change.

**Comment by jsntcy** [comment]
> > ❌AzureCLI-BreakingChangeTest
> 
> It has been verified with no breaking change.

Does it mean breaking change tool need fix these false positive? @AllyW

File: src/azure-cli/azure/cli/command_modules/vm/aaz/latest/vm/_reimage.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def _build_arguments_schema(cls, *args, **kwargs):
         )
         _args_schema.vm_name = AAZStrArg(
             options=["-n", "--name", "--vm-name"],
-            help="The name of the virtual machine.",
+            help="The name of the Virtual Machine. You can configure the default using `az configure --defaults vm=<name>`",
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/aaz/latest/vm/_reimage.py:56]: Is this a new feature automatically supported by Code Gen?

Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/aaz/latest/vm/_reimage.py:56]: It is a new feature of Code Gen V2, which supports setting default values by `az configure`

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**